contrib/haproxy/patches/use_env_flags.patch

```diff
@@ -0,0 +1,29 @@
+diff --git Makefile Makefile
+index a66c9b69c..64c44b85f 100644
+--- Makefile
++++ Makefile
+@@ -168,7 +168,7 @@ ARCH =
+ 
+ #### Toolchain options.
+ CC = cc
+-LD = $(CC)
++LD = cc
+ 
+ #### Debug flags (typically "-g").
+ # Those flags only feed CFLAGS so it is not mandatory to use this form.
+@@ -289,13 +289,13 @@ ARCH_FLAGS        = $(ARCH_FLAGS.$(ARCH))
+ # These CFLAGS contain general optimization options, CPU-specific optimizations
+ # and debug flags. They may be overridden by some distributions which prefer to
+ # set all of them at once instead of playing with the CPU and DEBUG variables.
+-CFLAGS = $(ARCH_FLAGS) $(CPU_CFLAGS) $(DEBUG_CFLAGS) $(SPEC_CFLAGS)
++CFLAGS := $(ARCH_FLAGS) $(CPU_CFLAGS) $(DEBUG_CFLAGS) $(SPEC_CFLAGS) $(CFLAGS)
```

Comment:
while here, remove ARCH_FLAGS from both and CPU_CFLAGS since they're just noise

will have to check aarch64 later